### PR TITLE
7288: include WithdrawalRequestPredeployAddress in genesis configuration

### PR DESCRIPTION
**Pull request by Matilda-Clerke**
## PR description
Allow configuration of WithdrawalRequestPredeployAddress in genesis configuration

## Fixed Issue(s)

https://github.com/hyperledger/besu/issues/7288

### Thanks for sending a pull request! Have you done the following?

- [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [x] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### Locally, you can run these tests to catch failures early:

- [x] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

**Review by siladu — APPROVED** [review]
LGTM.

Might be nice to add a startup log when the address is different to default, could be a separate PR.

**Comment by Gabriel-Trintinalia** [comment]
@macfarla Do you think we should have the same for Consolidation requests since we have it for deposits and withdrawals?

File: config/src/main/java/org/hyperledger/besu/config/JsonGenesisConfigOptions.java [hunk 1]
```diff
@@ -49,6 +49,8 @@ public class JsonGenesisConfigOptions implements GenesisConfigOptions {
   private static final String CHECKPOINT_CONFIG_KEY = "checkpoint";
   private static final String ZERO_BASE_FEE_KEY = "zerobasefee";
   private static final String FIXED_BASE_FEE_KEY = "fixedbasefee";
+  private static final String WITHDRAWAL_REQUEST_PREDEPLOY_ADDRESS_KEY =
+      "withdrawalrequestpredeployaddress";
```
Comment (macfarla) [inline config/src/main/java/org/hyperledger/besu/config/JsonGenesisConfigOptions.java:53]:
```suggestion
      "withdrawalrequestcontractaddress";
```

File: config/src/main/java/org/hyperledger/besu/config/JsonGenesisConfigOptions.java [hunk 2]
```diff
@@ -49,6 +49,8 @@ public class JsonGenesisConfigOptions implements GenesisConfigOptions {
   private static final String CHECKPOINT_CONFIG_KEY = "checkpoint";
   private static final String ZERO_BASE_FEE_KEY = "zerobasefee";
   private static final String FIXED_BASE_FEE_KEY = "fixedbasefee";
+  private static final String WITHDRAWAL_REQUEST_PREDEPLOY_ADDRESS_KEY =
```
Comment (macfarla) [inline config/src/main/java/org/hyperledger/besu/config/JsonGenesisConfigOptions.java:52]:
```suggestion
  private static final String WITHDRAWAL_REQUEST_CONTRACT_ADDRESS_KEY =
```

**Review by macfarla — COMMENTED** [review]
think it makes sense to use "withdrawal request contract" similar to "deposit contract" rather than "predeploy" unless there's a reason to keep that

**Comment by macfarla** [comment]
> @macfarla Do you think we should have the same for Consolidation requests since we have it for deposits and withdrawals?

yes i think so. created #7381

**Review by siladu — COMMENTED** [review]
Changelog needs moving from 24.7.1 section to "next release"

**Comment by Matilda-Clerke** [comment]
Good catch, thanks Simon. I didn't think of that

**Comment by fdtwd8vv45-sketch** [comment]
> ## PR description
> 
> Allow configuration of WithdrawalRequestPredeployAddress in genesis configuration
> 
> 
> 
> ## Fixed Issue(s)
> 
> 
> 
> 
> 
> https://github.com/hyperledger/besu/issues/7288
> 
> 
> 
> ### Thanks for sending a pull request! Have you done the following?
> 
> 
> 
> - [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
> 
> - [x] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
> 
> - [x] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
> 
> 
> 
> ### Locally, you can run these tests to catch failures early:
> 
> 
> 
> - [x] unit tests: `./gradlew build`
> 
> - [ ] acceptance tests: `./gradlew acceptanceTest`
> 
> - [ ] integration tests: `./gradlew integrationTest`
> 
> - [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`
> 
> 
> 
>

**Pull request merged**